docs/data/toolpad/core/components/use-dialogs/CustomDialogWithPayload.tsx

```diff
@@ -8,7 +8,7 @@ import DialogActions from '@mui/material/DialogActions';
 import TextField from '@mui/material/TextField';
 import Stack from '@mui/material/Stack';
 
-function MyCustomDialog({ payload, open, onClose }: DialogProps<string>) {
+function MyCustomDialog({ payload, open, onClose }: DialogProps<any>) {
```

Comment:
no, we need to advertise strong typing

```suggestion
function MyCustomDialog({ payload, open, onClose }: DialogProps<string>) {
```